app/assets/v2/js/cart-data.js

```diff
@@ -244,4 +245,72 @@ class CartData {
+      // Find the latestGrantsData entry with the same grant ID as this grant
+      const grantIndex = latestGrantsData.findIndex((item) => Number(item.id) === Number(grant.grant_id));
+      const latestGrantData = latestGrantsData[grantIndex];
+
+      // Update with the grant data from server
+      cartData[index].binance_payout_address = latestGrantData.binance_payout_address;
+      cartData[index].celo_payout_address = latestGrantData.celo_payout_address;
+      cartData[index].clr_round_num = latestGrantData.clr_round_num;
+      cartData[index].grant_admin_address = latestGrantData.admin_address;
+      cartData[index].grant_clr_prediction_curve = latestGrantData.clr_prediction_curve;
+      cartData[index].grant_contract_address = latestGrantData.contract_address;
+      cartData[index].grant_contract_version = latestGrantData.contract_version;
+      cartData[index].grant_donation_num_rounds = 1; // always 1, since recurring contributions are not supported
+      cartData[index].grant_id = Number(latestGrantData.id); // should already be a number, so this is an extra safety check
+      cartData[index].grant_image_css = latestGrantData.image_css;
+      cartData[index].grant_logo = latestGrantData.logo_url;
+      cartData[index].grant_slug = latestGrantData.slug;
+      cartData[index].grant_title = latestGrantData.title;
+      cartData[index].grant_token_address = latestGrantData.token_address;
+      cartData[index].grant_token_symbol = latestGrantData.token_symbol;
+      cartData[index].grant_url = latestGrantData.url;
+      cartData[index].harmony_payout_address = latestGrantData.harmony_payout_address;
+      cartData[index].is_clr_eligible = latestGrantData.is_clr_eligible;
+      cartData[index].is_on_team = latestGrantData.is_on_team;
+      cartData[index].kusama_payout_address = latestGrantData.kusama_payout_address;
+      cartData[index].polkadot_payout_address = latestGrantData.polkadot_payout_address;
+      cartData[index].rsk_payout_address = latestGrantData.rsk_payout_address;
+      cartData[index].tenants = latestGrantData.tenants;
+      cartData[index].zcash_payout_address = latestGrantData.zcash_payout_address;
+      cartData[index].zil_payout_address = latestGrantData.zil_payout_address;
+
+      // Other localStorage properties not updated above:
+      //   grant_donation_amount:     Existing localStorage value is preserved
+      //   grant_donation_clr_match:  Updated in cart.js `grantData` watcher
+      //   grant_donation_currency:   Existing localStorage value is preserved
+      //   payment_status:            Updated on-demand in cart.js `updatePaymentStatus()` method
+      //   token_local_id:            Existing localStorage value is preserved
+      //   txnid:                     Updated on-demand in cart.js `updatePaymentStatus()` method
+      //   uuid:                      TODO is this needed or can it be removed? Currently existing value is preserved
```

Comment:
What is the UUID used for? what is a grant’s UUID used for? A non-deterministic UUID is generated in `CartData.addToCart()`, but I don’t see it used anywhere. Found the PR it was introduced in case that helps: https://github.com/gitcoinco/web/pull/6902

In this section of code, should we:
1. Preserve the UUID when updating grant info, or
2. Update the UUID, or
3. Get rid of the grant UUIDs entirely

Seems 3 is the way to go, but want to make sure I’m not missing anything first

Reply:
@owocki / @zoek1 would you be able to comment on this ?

I see the reason it was brought up on https://github.com/gitcoinco/web/issues/6892 but not clear beyond that !

@mds1 -> if we aren't sure and we know currently it isn't updated , I'd lean towards preserving that just to avoid weird side effects. Let's wait to see what kevin / zoek respond